File: backend/library/app/schemas/library.py
import enum
import uuid
from datetime import datetime, timezone
from typing import Optional

from pydantic import BaseModel, Field

from app.schemas.enums import LibraryStatus, SourceType


def build_minio_key(user_id: uuid.UUID, item_id: uuid.UUID, file_type: str) -> str:
    return f"users/{user_id}/documents/{item_id}.{file_type}"


class LibraryResponseSchema(BaseModel):
    model_config = {"from_attributes": True}

    id: uuid.UUID
    knowledge_id: uuid.UUID

    source_type: SourceType
    original_ref: str
    title: Optional[str] = None
    status: LibraryStatus
    error_message: Optional[str] = None
    text_content: Optional[str] = None
    compressed_content: Optional[str] = None
    order: Optional[int] = None

    created_at: datetime
    updated_at: Optional[datetime] = None

class LibraryCreateSchema(BaseModel):
    title: str

File: frontend/src/types/library.ts
export interface LibraryResponse {
  id: string
  knowledge_id: string
  source_type: string
  original_ref: string
  title?: string
  status: string
  error_message?: string
  text_content?: string
  compressed_content?: string
  order?: number
  created_at: string
  updated_at?: string
}

export interface LibraryUrlCreate {
  url: string
  title?: string
}
